lib/new_relic/agent/instrumentation/rails_notifications/action_controller.rb

```diff
@@ -32,15 +32,19 @@
     NewRelic::Agent::Instrumentation::ActionControllerSubscriber \
       .subscribe(/^process_action.action_controller$/)
 
-    subs = %w[send_file
+    subs = %w[exist_fragment?
+      expire_fragment
+      halted_callback
+      read_fragment
+      redirect_to
       send_data
+      send_file
       send_stream
-      redirect_to
-      halted_callback
-      unpermitted_parameters]
+      write_fragment
+      unpermitted_parameters].map { |s| Regexp.escape(s) }
 
     # have to double escape period because its going from string -> regex
     NewRelic::Agent::Instrumentation::ActionControllerOtherSubscriber \
-      .subscribe(Regexp.new("^(#{subs.join('|')})\\.action_controller$"))
+      .subscribe(Regexp.new("^(?:#{subs.join('|')})\\.action_controller$"))
```

Comment:
What is the addition of `?:` doing as a non-regex pro?

Reply:
By default, wrapping part of the pattern in parens will cause the part of the match to be captured. By using `(?:)` instead of `()`, I'm telling both the parser and the future human reviewer that we don't actually intend to do any capturing with this regex and simply want to use the parens for grouping.